application/src/main/java/org/opentripplanner/graph_builder/module/osm/EdgeLevelInfoProcessor.java

```diff
@@ -0,0 +1,120 @@
+package org.opentripplanner.graph_builder.module.osm;
+
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssueStore;
+import org.opentripplanner.graph_builder.issues.CouldNotApplyMultiLevelInfoToWay;
+import org.opentripplanner.osm.model.OsmLevel;
+import org.opentripplanner.osm.model.OsmWay;
+import org.opentripplanner.service.streetdecorator.OsmStreetDecoratorRepository;
+import org.opentripplanner.service.streetdecorator.model.EdgeLevelInfo;
+import org.opentripplanner.service.streetdecorator.model.Level;
+import org.opentripplanner.service.streetdecorator.model.VertexLevelInfo;
+import org.opentripplanner.street.model.edge.Edge;
+
+/**
+ * Contains logic for storing edge level info in the
+ * {@link OsmStreetDecoratorRepository}.
+ */
+class EdgeLevelInfoProcessor {
+
+  private final DataImportIssueStore issueStore;
+  private final OsmStreetDecoratorRepository osmStreetDecoratorRepository;
+  private final boolean includeEdgeLevelInfo;
+
+  public EdgeLevelInfoProcessor(
+    DataImportIssueStore issueStore,
+    boolean includeEdgeLevelInfo,
+    OsmStreetDecoratorRepository osmStreetDecoratorRepository
+  ) {
+    this.issueStore = issueStore;
+    this.includeEdgeLevelInfo = includeEdgeLevelInfo;
```

Comment:
Using this here feels slightly wrong. Maybe we should create a EdgeLevelInfoProcessor interface and some NoopEdgeLevelInfoProcessor when `includeEdgeLevelInfo == false` and use this when `includeEdgeLevelInfo == true`. `getEdgeLevelInfo` could be a static method then (although maybe osmdb should be member variable here). I'm not sure if `#getEdgeLevelInfo` and `#storeLevelInfoForEdge` should even be in the same class. At least we should add javadoc to what they do.